napari/components/viewer_model.py

```diff
@@ -350,10 +360,7 @@ def _update_layers(self, *, layers=None):
             List of layers to update. If none provided updates all.
         """
         layers = layers or self.layers
-        for layer in layers:
-            layer._slice_dims(
-                self.dims.point, self.dims.ndisplay, self.dims.order
-            )
+        self._layer_slicer.slice_layers_async(layers, self.dims)
         position = list(self.cursor.position)
```

Comment:
When async is enabled the cursor position will be updated immediately. That seems consistent with the slider positions being updated immediately. But I think the change in position will cause a call to `Layer._get_value` to get the value under the cursor, which will use values from the old slice.

As this only affect async and is not a serious issue, I don't think we need to fix this here, but I'm making a note so I don't forget.

Reply:
That's fine I'd say, as long as the old slice is still being visualized. Once the new slice is ready, we can swap it out and it will update what `get_value` gives. What I think we should avoid is someone pointing their cursor somewhere and getting a value that is different from what they see.

Reply:
What will actually happen here?
Will it display: 
1. `<new position> <value from old slice @ new position>`
2. `<new position> <value from old slice @ old position>`
3. [not this one, but isn't this more correct?] `<old position> <value from old slice @ old position>`

Reply:
Why is 3 more correct? I would say 1 is more correct. If I'm seeing "red" under my cursors, I want to get "red" from my get value. This is in line with the small "consensus" we had on `Dims.current_step` (as linked in [NAP-4](https://napari.org/dev/naps/4-async-slicing.html#discussion)):

https://github.com/napari/napari/pull/4892/files#r935336654

Reply:
I am still trying to understand the exact combination of information that is displayed, specifically is it *consistent* with the image data.

I agree that if you see `red = 253` under your cursor that you should get `253` as the value under cursor. What I am trying to clarify is the coordinate that is being displayed with respect to the pixel value that is shown. I want to make sure that the coordinate corresponds to the index the retrieved `253`.

Reply:
So to figure out if we are in this situation we currently need to ask the layer slicer with `_find_existing_task`? Does it make sense to add a function to `Layer` that we can use to check if it is fully loaded/valid?

Reply:
I believe the status updates should go through the `ViewerModel` (because they're only probing for the active layer), so we can also check if a layer has pending slices there. Given that the long term goal (not in this PR) is to remove slice state from the layer, I'm definitely against introducing a new public method in `Layer` to allow this check. But open to other ideas.

Given that the status/value bug only occurs in the async slicing case (which is off by default in this PR), I don't think issue blocks this PR. In general, I think we could merge this with some known issues. My main goal in this PR is to get confirmed buy-in around the long term change this introduces of separating slicing and layers.

Reply:
Okeydoki. Fair enough. Restating:

cursor behavior with sync slicing will still have the correct behavior. async cursor will be incorrect during slicing. we will fix this but, because we aren't in a position to do this properly at the moment and the impact is minor relative to the value of enabling async, we will add the fix later.

Reply:
Nice; just to make sure, can we add a comment somewhere in the code that clarifies where/how this can break?

Reply:
Done!